src/internal/pfssync/cache_client.go

```diff
@@ -0,0 +1,59 @@
+package pfssync
+
+import (
+	io "io"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pfsdb"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/renew"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+type CacheClient struct {
+	*client.APIClient
+	mu      sync.Mutex
+	cache   *simplelru.LRU
+	renewer *renew.StringSet
+}
+
+// TODO: Expose configuration for cache size?
+// TODO: Dedupe work?
+func NewCacheClient(pachClient *client.APIClient, renewer *renew.StringSet) *CacheClient {
+	cc := &CacheClient{
+		APIClient: pachClient,
+		renewer:   renewer,
+	}
+	cache, err := simplelru.NewLRU(100, cc.onEvicted)
+	if err != nil {
+		// lru.NewWithEvict only errors for size < 1
+		panic(err)
+	}
+	cc.cache = cache
+	return cc
+}
+
+func (cc *CacheClient) GetFileTAR(commit *pfs.Commit, path string) (io.ReadCloser, error) {
+	key := pfsdb.CommitKey(commit)
+	cc.mu.Lock()
```

Comment:
These lock/unlocks are probably going to get screwed up at some point.  They look correct at the moment.  Maybe pull them out into get and put methods.  Then you could add type safety around the cache, do the casting there.